ravenpy/utilities/geo.py

```diff
@@ -0,0 +1,238 @@
+        " `pip install ravenpy[gis]` recipe or via Anaconda (`conda env -n ravenpy-env -f environment.yml`)"
+        " from the RavenPy repository source files."
+    )
+    raise ImportError(msg) from e
+
+RASTERIO_TIFF_COMPRESSION = "lzw"
+LOGGER = logging.getLogger("RavenPy")
+WGS84 = 4326
+
+
+def geom_transform(
+    geom: Union[GeometryCollection, shape],
+    source_crs: Union[str, int, CRS] = WGS84,
+    target_crs: Union[str, int, CRS] = None,
+) -> GeometryCollection:
+    """Change the projection of a geometry.
+
+    Assuming a geometry's coordinates are in a `source_crs`, compute the new coordinates under the `target_crs`.
+
+    Parameters
+    ----------
+    geom : Union[GeometryCollection, shape]
+      Source geometry.
+    source_crs : Union[str, int, CRS]
+      Projection identifier (proj4) for the source geometry, e.g. '+proj=longlat +datum=WGS84 +no_defs'.
+    target_crs : Union[str, int, CRS]
+      Projection identifier (proj4) for the target geometry.
+
+    Returns
+    -------
+    GeometryCollection
+      Reprojected geometry.
+    """
+    try:
+        from pyproj import Transformer  # noqa
+        from functools import partial
+
+        source = (
+            CRS.from_epsg(source_crs)
+            if isinstance(source_crs, int or str)
+            else source_crs
+        )
+        target = (
+            CRS.from_epsg(target_crs)
+            if isinstance(target_crs, int or str)
+            else target_crs
+        )
+
+        transform_func = Transformer.from_crs(source, target, always_xy=True)
+        reprojected = transform(transform_func.transform, geom)
+
+        return reprojected
+    except Exception as err:
+        msg = f"{err}: Failed to reproject geometry"
+        LOGGER.error(msg)
+        raise Exception(msg)
+
+
+def generic_raster_clip(
+    raster: Union[str, Path],
+    output: Union[str, Path],
+    geometry: Union[Polygon, MultiPolygon, List[Union[Polygon, MultiPolygon]]],
+    touches: bool = False,
+    fill_with_nodata: bool = True,
+    padded: bool = True,
+    raster_compression: str = RASTERIO_TIFF_COMPRESSION,
+) -> None:
+    """
+    Crop a raster file to a given geometry.
+
+    Parameters
+    ----------
+    raster : Union[str, Path]
+      Path to input raster.
+    output : Union[str, Path]
+      Path to output raster.
+    geometry : Union[Polygon, MultiPolygon, List[Union[Polygon, MultiPolygon]]
+      Geometry defining the region to crop.
+    touches : bool
+      Whether or not to include cells that intersect the geometry. Default: True.
+    fill_with_nodata: bool
+      Whether or not to keep pixel values for regions outside of shape or set as nodata. Default: True.
+    padded: bool
+      Whether or not to add a half-pixel buffer to shape before masking
+    raster_compression : str
+      Level of data compression. Default: 'lzw'.
+
+    Returns
+    -------
+    None
+    """
+    if not (type(geometry) in (list, tuple)):
```

Comment:
`not in` is an acceptable (and more readable IMO) python idiom.

But maybe even better: `if isinstance(geometry, Iterable):` ? (`collections.abc.Iterable`)